shared/src/Datadog.Trace.ClrProfiler.Native/process_helper.cpp

```diff
@@ -87,187 +90,60 @@ bool ProcessHelper::RunProcess(const std::string& processPath,
-                success = false;
-            }
+            _exit(EXIT_FAILURE);
         }
-        CloseIfOpen(waitForChildToExecPipe[READ_END_OF_PIPE]);
-    }
 
-    // If we failed, close everything else and give back error values in all out arguments.
-    if (!success)
-    {
-        Log::Warn("SingleStepGuardRails::SendTelemetry: Error calling telemetry forwarder");
-        // Reap child
-        if (processId > 0)
-        {
-            int status;
-            waitpid(processId, &status, 0);
-        }
-    }
+        close (read_fd); // Close the original read end of the pipe
 
-    return success;
-}
+        // Spawn the executable
+        int error = execve(processPath.c_str(), const_cast<char* const *>(argv.data()), environ);
 
-VoidIntFn ProcessHelper::handler_from_sigaction (struct sigaction *sa)
-{
-    if (((unsigned int)sa->sa_flags) & SA_SIGINFO)
-    {
-        // work around -Wcast-function-type
-        void (*tmp)(void) = (void (*)(void))sa->sa_sigaction;
-        return (void (*)(int))tmp;
+        // execve should take over the execution.
+        // If we get here, it means execve failed, so we tear down the process.
+        _exit(error != 0 ? error : EXIT_FAILURE);
     }
-    else
-    {
-        return sa->sa_handler;
-    }
-}
 
-void ProcessHelper::CloseIfOpen(int fd)
-{
-    if (fd >= 0)
-    {
-        close(fd); // Ignoring errors from close is a deliberate choice
-    }
-}
-
-inline bool ProcessHelper::CheckInterrupted(ssize_t result)
-{
-    return result < 0 && errno == EINTR;
-}
+    // This part is only executed by the parent process
 
-ssize_t ProcessHelper::WriteSize(int fd, const void* buffer, size_t count)
-{
-    ssize_t rv = 0;
-    while (count > 0)
+    if (processId > 0)
     {
-        ssize_t result = 0;
-        while (CheckInterrupted(result = write(fd, buffer, count)));
-        if (result > 0)
-        {
-            rv += result;
-            buffer = (const uint8_t*)buffer + result;
-            count -= (size_t)result;
-        }
-        else
-        {
-            return -1;
-        }
-    }
-    return rv;
-}
+        close(read_fd); // The parent process doesn't need to read
 
-ssize_t ProcessHelper::ReadSize(int fd, void* buffer, size_t count)
-{
-    ssize_t rv = 0;
-    while (count > 0)
-    {
-        ssize_t result = 0;
-        while (CheckInterrupted(result = read(fd, buffer, count)));
-        if (result > 0)
-        {
-            rv += result;
-            buffer = (uint8_t*)buffer + result;
-            count -= (size_t)result;
-        }
-        else
-        {
-            return -1;
-        }
+        write(write_fd, input.data(), input.length());
+        close(write_fd);
     }
-    return rv;
-}
-
-void ProcessHelper::ExitChild(int pipeToParent, int error)
-{
-    if (pipeToParent != -1)
+    else
     {
-        WriteSize(pipeToParent, &error, sizeof(error));
+        Log::Warn("SingleStepGuardRails::SendTelemetry: Error calling telemetry forwarder");
```

Comment:
Shouldn't we return here, if processId <= 0? Because we don't want to call `waitpid` on it right?

```suggestion
        Log::Warn("ProcessHelper::RunProcess: Error starting ", processPath);
        return false;
```